lib/rubocop/cop/ezcater/graphql_fields_naming.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Ezcater
+      # This cop makes sure that GraphQL fields & arguments match the supported style
+      # https://git.io/JeofW
+      #
+      # The cop also ignores when users provide a :camelize option, because
+      # the user is manually requesting to override the value
+      #
+      # @example
+      #   # bad
+      #   field :fooBar, ID, null: false
+      #   argument :barBaz, ID, required: true
+      #
+      #   # good
+      #   field :foo_bar, ID, null: true
+      #   field :foo_bar, ID, null: true do
+      #     argument :bar_baz, ID, required: true
+      #   end
+      #
+      #   field :fooBar, ID, null: true, camelize: true
+      #
+      class GraphqlFieldsNaming < Cop
+        include ConfigurableNaming
+
+        MSG = "Use %<style>s for GraphQL fields & arguments names. " \
+              "See https://git.io/JeofW for our guide. " \
+              "This can be overridden with camelize."
+        FIELD_ADDING_METHODS = %i(field argument).freeze
+        PROCESSABLE_TYPES = %i(sym string).freeze
+
+        def on_send(node)
```

Comment:
The method definition for fields can be seen here: https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/member/has_fields.rb#L9-L16